attestation-evidence-broker/src/aeb_modules/sev_aeb/get_evidence.rs

```diff
@@ -0,0 +1,157 @@
+use anyhow::*;
+use codicon::Decoder;
+use rand::Rng;
+use sev::{
+    certs::{
+        ca,
+        sev::{Certificate, Chain, Usage},
+    },
+    firmware::{AttestationReport, Firmware},
+    Generation,
+};
+use std::time::Duration;
+
+use crate::aeb_modules::sev_aeb::evidence::SevEvidence;
+
+fn download(url: &str, usage: Usage) -> Result<Certificate> {
+    let mut rsp = reqwest::blocking::get(url);
+    let mut http_request_replies = Vec::new();
+
+    // The AMD KDS server only accepts requests every 10 seconds
+    for attempt in 1..4 {
+        match &rsp {
+            // HTTP request has succeeded, ensure that the status code does not indicate an error.
+            std::result::Result::Ok(found) => {
+                if found.status().is_success() {
+                    break;
+                } else {
+                    http_request_replies.push(format!(
+                        "Attempt #{}, Error: Received HTTP response #{}",
+                        attempt,
+                        found.status()
+                    ));
+
+                    if let Some(retry) = found.headers().get("retry-after") {
+                        std::thread::sleep(Duration::from_secs(retry.to_str()?.parse()?));
```

Comment:
If the "?" invoke returns error, this function will return error directly with corresponding error info. But this function's return value type is  "Result<Certificate>", it doesn't return error info.
So I don't sure whether potential conflicts will happen here.